src/code42cli/cmds/devices.py

```diff
@@ -330,13 +342,61 @@ def list_devices(
         df = _add_settings_to_dataframe(state.sdk, df)
     if include_usernames:
         df = _add_usernames_to_device_dataframe(state.sdk, df)
+    if include_legal_hold_membership:
+        df = _add_legal_hold_membership_to_device_dataframe(state.sdk, df)
     if df.empty:
         click.echo("No results found.")
     else:
         formatter = DataFrameOutputFormatter(format)
         formatter.echo_formatted_dataframe(df)
 
 
+def _add_legal_hold_membership_to_device_dataframe(sdk, df):
+    matters = _get_all_active_matters(sdk)
```

Comment:
I agree with Kiran above that we probably shouldn't use the legal-hold module's internal functions, just for the sake of if someone later needs to change them it could break stuff here. 

Since we only need to iterate through all membership dicts, we can simplify getting them all with a generator function instead of building a new list and extending it: 

```
def _get_all_active_hold_memberships(sdk):
    for page in sdk.legalhold.get_all_matters(active=True):
        for matter in page["legalHolds"]:
            for _page in sdk.legalhold.get_all_matter_custodians(
                legal_hold_uid=matter["legalHoldUid"], active=True
            ):
                for membership in _page["legalHoldMemberships"]:
                    yield membership
```

And pandas has a great function that can 'flatten' nested json into a dataframe for you, which can simplify a lot of this logic: 

```
>>> df = pandas.json_normalize(_get_all_active_hold_memberships(sdk))
>>> df.head()
  legalHoldMembershipUid  active                   creationDate legalHold.legalHoldUid legalHold.name        user.userUid                   user.username                      user.email user.userExtRef
0     955061003769628092    True  2020-05-20T15:58:38.823-05:00     955060991270602172          Test2  945056771151950748                test@example.com                test@example.com            None
1     988814086101244865    True  2021-01-08T11:25:23.645-06:00     946193444244471969           Test  973619347957360307                legalhold@code42.com            legalhold@code42.com        None
2     955320624671252917    True  2020-05-22T10:57:44.903-05:00     946193444244471969           Test  955163198918558214                de60@example.com                de60@example.com            None
3     955320624637698485    True  2020-05-22T10:57:44.892-05:00     946193444244471969           Test  955163197341499910                de59@example.com                de59@example.com            None
4     955320624604144053    True  2020-05-22T10:57:44.885-05:00     946193444244471969           Test  955163195865104902                de58@example.com                de58@example.com            None
```

Reply:
`json_normalize` is great for unpacking the nested dictionary. However, I seem to be losing records (1 per matter). If I run the generator response table through `pandas.DataFrame`, I get a table with more rows than I do running the same generator response through `json_normalize`. I haven't quite worked this out yet

Reply:
Interesting, I'll see if I get the same behavior and try to figure out what's happening too.

Reply:
I had to serialize the generator into a list in order for all records to be included, which may defeat the purpose of the generator.

Reply:
Looks like there was a recent pandas bug about the generator thing (they weren't expecting generators to be passed): https://github.com/pandas-dev/pandas/issues/35923

It's fixed in 1.2.2, but I think we can probably just call list() on the result for now instead of force updating to the latest pandas.